### ISSUE TITLE
Beagle not working on Eclipse Kepler SR2 [4.3.2] 32bit Windows

### ISSUE THREAD

**Issue by ghost** [qos-ch/logback-beagle#2]
Hello,

I installed Logback Beagle on a fresh Eclipse Kepler SR2 23-bit on Windows 7 and it does not show up in the Show view window. There is no Logback category.

Eclipse Nebula Grid version: 1.0.0.201403032113

Thanks in advance !

**Comment by dpatriarche** [comment]
I have a similar problem on Mac: I installed Beagle following the instructions [here](http://logback.qos.ch/beagle/index.html), restarted Eclipse, and while the plugins are listed as installed Beagle doesn't show up in the list views or in the global Preferences.

Versions:
OS X 10.9.3
eclipse-java-kepler-SR2-macosx-cocoa-x86_64
Eclipse plugin versions:
- Nebula Grid: 1.0.0.201405151353
- SLF4J: 1.6.6
- Logback: 1.0.7
- Logback Beagle: 1.1.4.201209042005

**Comment by StevenP94** [comment]
No chance to recompile the plugin to update references ?

**Comment by thomasgillet** [comment]
Hello,

I just made Beagle work in Eclipse Kepler. As suggested in the [Jira bug](http://jira.qos.ch/browse/CONSPLUG-44), it was a problem with the SLF4J API bundle, which changed from `slf4j.api-1.6.6` to `org.slf4j.api-1.7.2` in the Kepler release. So the `Require-Bundle` manifest header must be changed accordingly in the `ch.qos.logback.beagle` bundle.
Eclipse Luna is still using the same SLF4J bundle so it should work as well.

Commit: d7d8f51a96ddffa6c64c7f99fd8f5d84a6f727d8

If you don't want to recompile the plugin, you can modify the manifest in the JAR file with any ZIP utility: install Beagle from Eclipse and then modify the JAR in the `plugins` directory (just don't forget to restart Eclipse with the `-clean` option afterward).

EDIT: see my comment below for a detailed explanation.

**Comment by kleini** [comment]
Why don't you specify a minimum required version? This will break again with an upgrade of slf4j.api. Furthermore your fix does not work with Eclipse versions prior to Kepler.

Second problem is, that you use Require-Bundle. This fails again if the bundle name is changed; what happened in this case, too. Normally it is advised to use Import-Package.

Use
Import-Package: org.slf4j;version="[1.6.6,2.0.0)"
and the problem would not have been occured and should not occur with Eclipse Mars again.

**Comment by thomasgillet** [comment]
Yes indeed, the `Require-Bundle` header is a bad practice in OSGi (and an even worse one when using a fixed version number). The `Import-Package` header should be preferred for a long-term solution, as you suggested.
But this is a wider issue, which should be addressed for all dependencies, not only SLF4J. (And this may not be an easy issue if you don't have the right tool - see OSGi semantic versioning.)

**Comment by kleini** [comment]
Why not fixing that part you are changing anyhow to the most correct solution? Why do you need tooling to write
Bundle-Import: org.slf4j;version="[1.6.6,2.0.0)"?

**Comment by StevenP94** [comment]
I've updated the required bundle like thomasgillet suggest (thx for the hint) but still no 'Logback' section is available on the Show View into Eclipse Luna. Any idea ?

**Comment by thomasgillet** [comment]
Hello @StevenP94,

I just installed Beagle in a new Luna instance (4.4.0-20140612-0600) and it works fine.
See below my exact steps, maybe it can help (first part is copied from http://logback.qos.ch/beagle/).
- Install the `Nebula Grid Feature` feature from http://download.eclipse.org/technology/nebula/snapshot/
- Install the `Logback Beagle` feature from http://logback.qos.ch/p2/
  _Do not_ take it from Eclipse Marketplace, its an older version.
  _Do not_ install other features from this repo (the beagle site screenshot shows all 3 features checked, but don't do it).
- Close Eclipse
- Make appropriate changes in `{eclipse-dir}/plugins/ch.qos.logback.beagle_1.1.4.201209042005.jar`
- Start Eclipse (I didn't even need the `-clean` option this time)

And then the Beagle view should be available.
If you do not see it, open the `Installation Details` dialog, go to the `Plug-ins` tab and search for "beagle". One plugin should appear if Beagle was correctly started.

**Comment by dlealv** [comment]
Thanks I have followed the same step, but it doesn't work...what do you mean with: Make appropriated changes in {eclipse-dir}...

Thanks in advance,

David

**Comment by thomasgillet** [comment]
Well, as said in my very first post, I mean make changes to the manifest file of the beagle JAR according to this: d7d8f51a96ddffa6c64c7f99fd8f5d84a6f727d8

**Comment by TowerKing** [comment]
@StevenP94 maybe you just modify the bundle-version to 1.7.2, not add org. before slf4j.
Look carefully the commit [d7d8f51](https://github.com/qos-ch/logback-beagle/commit/d7d8f51a96ddffa6c64c7f99fd8f5d84a6f727d8)

**Comment by berhauz** [comment]
Thanks, the detailed procedure above also works on Spring Tool Suite 3.6 and upgraded to 3.7 (based on Eclipse Luna SR2 (4.4.2)); yet I had to restart with "sts.exe -clean" to clean-up a previous 'beagle' installation that did not match exactly the steps above (I removed all beagle plug-ins as well as slf4j 1.6.6. and restarted).

**Comment by brainbytes42** [comment]
Hi,
thanks for the input, but on Eclipse 4.5.2 I didn't manage to get Beagle run, even with clean-option.
Too bad, as the Plugin looks really nice.
